src/tests/unit/doughs-dao.test.js

```diff
@@ -0,0 +1,100 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+
+import { getDoughsData } from './test-data';
+
+chai.should();
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+
+describe('test doughs dao', () => {
+  let serializeDoughsStub;
+  let getConnectionStub;
+  let connectionSpy;
+  let doughsDao;
+  beforeEach(() => {
+    connectionSpy = sinon.stub().returns(getDoughsData.executeReturn);
+    serializeDoughsStub = sinon.stub().returns(getDoughsData.baseGetDoughsReturn);
+    getConnectionStub = sinon.stub().resolves({
+      execute: connectionSpy,
+      close: () => { },
+      commit: () => { },
+    });
+
+    doughsDao = proxyquire('../../api/v1/db/oracledb/doughs-dao', {
+      './connection': {
+        getConnection: getConnectionStub,
+      },
+      '../../serializers/doughs-serializer': {
+        serializeDoughs: serializeDoughsStub,
+      },
+    });
+  });
+  afterEach(() => {
+    sinon.restore();
+  });
+  describe('getDoughs', () => {
+    it('returns the output from serializeDoughs', () => {
+      const result = doughsDao.getDoughs({});
+      return result.should.eventually.deep.equal(getDoughsData.baseGetDoughsReturn);
+    });
+    it('extracts only the rows from the database return', async () => {
+      await doughsDao.getDoughs({});
+      serializeDoughsStub.getCall(0).calledWith(getDoughsData.executeReturnRows).should.be.true;
+    });
+    describe('when it has an invalid filter', () => {
+      it('does not parse the invalid filter', async () => {
+        await doughsDao.getDoughs(getDoughsData.invalidFilters);
+        connectionSpy
+          .getCall(0)
+          .should.have.been
+          .calledWith(
+            getDoughsData.getDoughsQuery,
+            getDoughsData.emptyBindParams,
+          );
```

Comment:
I could see the argument for that. My thought originally was that if our tests show that it returns the output of serializeDoughs under two different cases it wouldn't be necessary to test all of them, but I'm not really committed to that. Maybe I should refactor the tests for this to iterate over a bunch of different input options and test each one to see if it still returns the right input.

Reply:
The updated lines 55-66 have some logic to loop over all the different input filter categories and see if it still returns the output of serializeDoughs. Let me know what you think about that solution.

Reply:
Looks good. The only suggestion is you can destructure the test data at once instead of accessing via getDoughsData every time since they're used many times during this unit test.

Reply:
Thanks, good suggestion